Bottom: ad3f7a0a6f6c13968275b2bfbd5f06c351ccb38c
Top:    6292aeac790956d1e4b0a2b4b867a089550c076d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-27 13:33:23 +0100

Move some more GDK routines away


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 50ac011..c6172e3 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -887,6 +887,28 @@ ghid_show_crosshair (gboolean show)
 }
 
 void
+ghid_drawing_area_configure_hook (GHidPort *port)
+{
+  static int done_once = 0;
+
+  if (!done_once)
+    {
+      port->bg_gc = gdk_gc_new (port->drawable);
+      gdk_gc_set_foreground (port->bg_gc, &port->bg_color);
+
+      port->offlimits_gc = gdk_gc_new (port->drawable);
+      gdk_gc_set_foreground (port->offlimits_gc, &port->offlimits_color);
+      done_once = 1;
+    }
+
+  if (port->mask)
+    {
+      gdk_pixmap_unref (port->mask);
+      port->mask = gdk_pixmap_new (0, port->width, port->height, 1);
+    }
+}
+
+void
 ghid_screen_update (void)
 {
   ghid_show_crosshair (FALSE);
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 525dd93..18c0b73 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -514,6 +514,7 @@ void ghid_fill_rect (hidGC gc, int x1, int y1, int x2, int y2);
 void ghid_invalidate_lr ();
 void ghid_invalidate_all ();
 void ghid_show_crosshair (gboolean show);
+void ghid_drawing_area_configure_hook (GHidPort *port);
 void ghid_screen_update (void);
 gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
                                       GHidPort *);
